player/command.c

```diff
@@ -5540,6 +5540,21 @@ static void cmd_expand_path(void *p)
     };
 }
 
+static void cmd_escape_ass(void *p)
+{
+    struct mp_cmd_ctx *cmd = p;
+    bstr dst = {0};
+
+    osd_mangle_ass(&dst, cmd->args[0].v.s, true);
+    if (!dst.start)
```

Comment:
I think would be clearer to use `bstr_xappend` instead of `MP_TARRAY_APPEND`. Similar result, but more readable maybe.

Reply:
or maybe in fact it is fine, because it makes it more direct in what it does. bstr would add two `\0`. So maybe no need to change.